src/transformers/models/dac/modeling_dac.py

```diff
@@ -0,0 +1,665 @@
+                Quantized continuous representation of input.
+            audio_codes (`torch.Tensor` of shape `(batch_size, num_codebooks, time_steps)`, *optional*):
+                The codebook indices for each codebook, representing the quantized discrete
+                representation of the input. This parameter should be provided if you want
+                to decode directly from the audio codes (it will overwrite quantized_representation).
+            return_dict (`bool`, *optional*):
+                Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+        Returns:
+            audio_values (`torch.Tensor` of shape `(batch_size, input_length)`):
+                Decoded audio data.
+        """
+
+        if quantized_representation is None and audio_codes is None:
+            raise ValueError("Either `quantized_representation` or `audio_codes` must be provided.")
+
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+
+        if audio_codes is not None:
+            quantized_representation = self.quantizer.from_codes(audio_codes)[0]
+
+        audio_values = self.decoder(quantized_representation).squeeze(1)
+
+        if not return_dict:
+            return (audio_values,)
+
+        return DacDecoderOutput(audio_values)
+
+    @add_start_docstrings_to_model_forward(DAC_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=DacOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        input_values: torch.Tensor,
+        n_quantizers: int = None,
+        return_dict: Optional[bool] = None,
+    ):
+        """
+        Returns:
+            loss (`torch.Tensor` of shape `(batch_size, 1)`):
+                Loss from the encoder model, comprising the weighted combination of the commitment and codebook losses.
+            audio_values (`torch.Tensor` of shape `(batch_size, input_length)`):
+                Decoded audio data.
+            quantized_representation (`torch.Tensor` of shape `(batch_size, dimension, time_steps)`):
+                Quantized continuous representation of input.
+            audio_codes: (`torch.Tensor` of shape `(batch_size, num_codebooks, time_steps)`):
+                Codebook indices for each codebook (quantized discrete representation of input).
+            projected_latents (torch.Tensor` of shape `(batch_size, num_codebooks * dimension, time_steps)`):
+                Projected latents (continuous representation of input before quantization),
+                shape .
+        """
+
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+        length = input_values.shape[-1]
+        loss, quantized_representation, audio_codes, projected_latents = self.encode(
+            input_values, n_quantizers, return_dict=False
+        )
+        audio_values = self.decode(quantized_representation, return_dict=False)
+
+        if isinstance(audio_values, tuple):
```

Comment:
Why would `audio_values` be a tuple here?

Reply:
The output to `decode` will be a tuple if return_dict is set to False. Actually in the `forward` method `audio_values` is always a tuple! (no need to check if it's a tuple or not... I'll update the code accordingly).